File: regexlib/2021-5-15/javascript_test_file/regexlib_6998.js
/* 6998
 * .*[\$Ss]pecia[l1]\W[Oo0]ffer.*
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"$pecial$Offer"*5000+"! _1SLQ_2\n"
 */
var REGEX = /.*[\$Ss]pecia[l1]\W[Oo0]ffer.*/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '$pecial$Offer'.repeat(i*10000) + '! _1SLQ_2\n'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/python_re_test_file/regexlib_6998.py
# 6998
# .*[\$Ss]pecia[l1]\W[Oo0]ffer.*
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"$pecial$Offer"*5000+"! _1SLQ_2\n"

import re
from time import perf_counter

regex = """.*[\$Ss]pecia[l1]\W[Oo0]ffer.*"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "$pecial$Offer" * i * 10000 + "! _1SLQ_2\n"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")